### PR TITLE
feat: improved installation mechanism

### LINKED ISSUE #443: [FEAT] add rebase script akin to the iso script that interactively prompts the user and rebases according to their answers
it should also ensure that the installed system is fedora atomic

### LINKED ISSUE #563: [DOCS] consider dropping the iso script
Maintaining support for user-built images via an unofficial iso builder means we're sending new users down two different installation pathways.

Instead, we should make rebasing easier by condensing it to one script:

related: https://github.com/secureblue/secureblue/issues/443

### PR DESCRIPTION
**Pull request by RoyalOughtness**

### CODE REVIEW

File: files/justfiles/secureblue.just [hunk 1]
```diff
@@ -79,6 +79,12 @@ remove-kargs-hardening:
       --delete-if-present="ia32_emulation=0"
     echo "Hardening kargs removed."
 
+# Ensure the signed image is being used
+switch-to-signed:
```
Comment (RoyalOughtness) [inline files/justfiles/secureblue.just]: this is no longer needed as of https://github.com/secureblue/secureblue/pull/564/files#diff-d48867bc4fd1881c03a2b2fee68a985512e7633905af8ea05fd5f62ccb8021ee

File: files/system/usr/share/ublue-os/firstboot/yafti.yml [hunk 2]
```diff
@@ -19,6 +19,15 @@ screens:
       actions:
         - run: ujust harden-flatpak
 
+  can-we-switch-to-signed:
```
Comment (RoyalOughtness) [inline files/system/usr/share/ublue-os/firstboot/yafti.yml]: neither is this https://github.com/secureblue/secureblue/pull/564/files#diff-d48867bc4fd1881c03a2b2fee68a985512e7633905af8ea05fd5f62ccb8021ee

File: install_secureblue.sh [hunk 3]
```diff
@@ -0,0 +1,82 @@
...
+            break
+        else
+            echo "Invalid option, please select a valid number."
+        fi
+    done
+    
+fi
+
+# Ask about Nvidia for all options
+read -p "Do you use Nvidia? (yes/No): " use_nvidia
```
Comment (RoyalOughtness) [inline install_secureblue.sh]: needs nvidia-open as well

File: docs/README.md [hunk 4]
```diff
@@ -77,160 +77,24 @@ Sponsorship options are on the [Donate](DONATE.md) page. All donations are appre
...
-- Then rebase to the signed image, like so:
-  ```
-  rpm-ostree rebase ostree-image-signed:docker://ghcr.io/secureblue/IMAGE_NAME:latest
-  ```
-- Reboot again to complete the installation:
-  ```
-  systemctl reboot
-  ```
-<sup>‡</sup> Replace `IMAGE_NAME` with the *full name* of your preferred image from the [list below](README.md#images).
+Then, mark it as executable (Right click -> Properties):
```
Comment (spaceoden) [inline docs/README.md]: May not apply to other DEs. And this is not needed if running `bash install_secureblue.sh`, as users don't need a file to be marked as executable in order to directly command bash to run said file.

File: docs/README.md [hunk 5]
```diff
@@ -77,160 +77,24 @@ Sponsorship options are on the [Donate](DONATE.md) page. All donations are appre
...
 
-## ISO
+![image](https://github.com/user-attachments/assets/7c2f203e-0772-417f-b1e0-2f5db5fd1343)
 
-While it's recommended to use a Fedora Atomic ISO to install and then rebase that installation to secureblue, you can also generate an ISO and install that directly using [this script](generate_secureblue_iso.sh). Please note you should still follow the [post-install steps](README.md#post-install) when installing from a generated ISO:
+Then, run it: (Right click -> Run as a Program):
```
Comment (spaceoden) [inline docs/README.md]: this may not apply to other DEs

**Review by spaceoden — COMMENTED** [review]
the file manager based instructions are going to vary based on DE. It might be better to direct users to simply download the script and run `bash install_secureblue.sh` (don't need to mark executable when specifying bash). We could also direct them to use curl or wget to download the file. And we could provide both the GUI (browser and file manager) and terminal instructions if deemed helpful.

File: docs/README.md (hunk 4 above)
Comment (RoyalOughtness) [reply docs/README.md]: good point

**Comment by RoyalOughtness** [comment]
closing while i write tests to prevent github actions spam

**Review by TheHooly — APPROVED** [review]

**Review by mkkvcs — APPROVED** [review]

**Comment by tzugen** [comment]
Hey, the previous README mentioned the need to do a second rebase, that doesnt seem to be handled by the new script?

**Comment by RoyalOughtness** [comment]
> Hey, the previous README mentioned the need to do a second rebase, that doesnt seem to be handled by the new script?

it's handled automatically by a firstrun service

**Pull request merged**